lib/web/apiserver_test.go

```diff
@@ -6482,3 +6488,156 @@ func init() {
 		&metav1.Status{},
 	)
 }
+
+// TestForwardingTraces checks that the userContext includes the ID of the
+// access request after it has been consumed and the web session has been renewed.
+func TestForwardingTraces(t *testing.T) {
+	t.Parallel()
+
+	env := newWebPack(t, 1)
+	p := env.proxies[0]
+
+	span := &tracepb.TracesData{
+		ResourceSpans: []*tracepb.ResourceSpans{
+			{
+				Resource: &resourcev1.Resource{
+					Attributes: []*commonv1.KeyValue{
+						{
+							Key: "test",
+							Value: &commonv1.AnyValue{
+								Value: &commonv1.AnyValue_IntValue{
+									IntValue: 0,
+								},
+							},
+						},
+					},
+				},
+				ScopeSpans: []*tracepb.ScopeSpans{
+					{
+						Spans: []*tracepb.Span{
+							{
+								TraceId:           []byte{1, 2, 3, 4},
+								SpanId:            []byte{5, 6, 7, 8},
+								TraceState:        "",
+								ParentSpanId:      []byte{9, 10, 11, 12},
+								Name:              "test",
+								Kind:              tracepb.Span_SPAN_KIND_CLIENT,
+								StartTimeUnixNano: uint64(time.Now().Add(-1 * time.Minute).Unix()),
+								EndTimeUnixNano:   uint64(time.Now().Unix()),
+								Attributes: []*commonv1.KeyValue{
+									{
+										Key: "test",
+										Value: &commonv1.AnyValue{
+											Value: &commonv1.AnyValue_IntValue{
+												IntValue: 0,
+											},
+										},
+									},
+								},
+								Status: &tracepb.Status{
+									Message: "success!",
+									Code:    tracepb.Status_STATUS_CODE_OK,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	newRequest := func(t *testing.T) *http.Request {
+		req, err := http.NewRequest(http.MethodGet, "", nil)
+		require.NoError(t, err)
+
+		return req
+	}
+
+	cases := []struct {
+		name      string
+		req       func(t *testing.T) *http.Request
+		assertion func(t *testing.T, spans []*otlp.ResourceSpans, err error)
+	}{
+		{
+			name: "no traces",
+			req: func(t *testing.T) *http.Request {
+				r := newRequest(t)
+
+				raw, err := protojson.Marshal(&tracepb.ResourceSpans{})
+				require.NoError(t, err)
+				r.Body = io.NopCloser(bytes.NewBuffer(raw))
+
+				return r
+			},
+			assertion: func(t *testing.T, spans []*tracepb.ResourceSpans, err error) {
+				require.NoError(t, err)
+				require.Empty(t, spans)
```

Comment:
```suggestion
				require.Empty(t, spans, "Spans mismatch")
```

Same for others. (Message could be better, up to you.)